open_spiel/python/games/liars_poker.py

```diff
@@ -0,0 +1,379 @@
+    count = bid % (self._hand_length * self._num_players)
+    number = self._deck[bid // (self._hand_length * self._num_players)]
+    return (count, number)
+
+  def _end_game(self):
+    """Ends the game by calling a counts and setting respective attributes."""
+    self._counts()
+    self._game_over = True
+
+  def _counts(self):
+    """
+    Determines if the bid originator wins or loses.
+    """
+    bid_count, bid_number = self._decode_bid(self._current_bid - self._bid_offset)
+
+    # Count the number of bid_numbers from all players.
+    matches = 0
+    for player_id in range(self._num_players):
+      for digit in self.hands[player_id]:
+        if digit == bid_number:
+          matches += 1
+    
+    # If the number of matches are at least the bid_count bid, then the bidder wins.
+    # Otherwise everyone else wins.
+    if matches >= bid_count:
+      self._winner = self._bid_originator
+    else:
+      self._loser = self._bid_originator
+
+  def _update_bid_history(self, bid, player):
+    """Writes a player's bid into memory."""
+    self.bid_history[bid][player] = 1
+
+  def _update_challenge_history(self, bid, player):
+    """Write a player's challenge for a bid into memory."""
+    self.challenge_history[bid][player] = 1
+
+  def _apply_action(self, action):
+    """Applies an action and updates the state."""
+    if self.is_chance_node():
+      # If we are still populating hands, draw a number for the current player.
+      self.hands[self._current_player].append(action)
+    elif action == Action.CHALLENGE:
+      assert self._is_challenge_possible()
+      self._update_challenge_history(
+        self._current_bid - self._bid_offset, self._current_player)
+      self._num_challenges += 1
+      # If there is no ongoing rebid, check if all players challenge before counting.
+      # If there is an ongoing rebid, count once all the players except the bidder challenges.
+      if (not self.is_rebid and self._num_challenges == self._num_players) or (
+        self.is_rebid and self._num_challenges == self._num_players - 1):
+        self._end_game()
+    else:
+      # Set the current bid to the action.
+      self._current_bid = action
+      if self._current_player == self._bid_originator:
+        # If the bid originator is bidding again, we have a rebid.
+        self.is_rebid = True
+      else:
+         # Otherwise, we have a regular bid.
+         self.is_rebid = False
+      # Set the bid originator to the current player.
+      self._bid_originator = self._current_player
+      self._update_bid_history(self._current_bid - self._bid_offset, self._current_player)
+      self._num_challenges = 0
+    self._current_player = (self._current_player + 1) % self._num_players
+
+  def _action_to_string(self, player, action):
+    """Action -> string."""
+    if player == pyspiel.PlayerId.CHANCE:
+      return f"Deal:{action}"
+    elif action == Action.CHALLENGE:
+      return "Challenge"
+    else:
+      return "Bet"
```

Comment:
(see also comment in playthrough):
Different action id's should have different strings, so probably an amount /count needs to be added to Bet